### PR TITLE
feat(leadspace): add shadow parts

### PR DESCRIPTION
**Pull request by Valentin-Sorin-Nicolae**
[ADCMS-5161](https://jsw.ibm.com/browse/ADCMS-5161)

Description

All non-slot elements in the shadow DOM should be given a unique "part" name allowing CSS to target and override component default styles. This is for the "leadspace" component.

Changelog

New

Adding the shadow parts for the "leadspace" component and documentation

### CODE REVIEW

File: packages/web-components/src/components/leadspace/leadspace.ts [hunk 1]
```diff
@@ -197,21 +206,27 @@ class C4DLeadSpace extends StableSelectorMixin(LitElement) {
                       }
                     </linearGradient>
                   </defs>
-                  <rect class="${c4dPrefix}--leadspace__gradient__rect" width="100" height="100" />
+                  <rect class="${c4dPrefix}--leadspace__gradient__rect" part="gradient-rect" width="100" height="100" />
```
Comment (andy-blum) [inline packages/web-components/src/components/leadspace/leadspace.ts:209]: I think we probably ought to leave the gradients alone here. Thoughts @m4olivei?

Comment (m4olivei) [reply packages/web-components/src/components/leadspace/leadspace.ts:209]: It does seem a bit overkill. I can't see someone wanting to target parts of the SVG. There are plenty of other SVG icons in components where we wouldn't easily have the ability to add parts to the child elements of the SVG, only the SVG tag itself. My vote would be to just have a `part` attribute on the `<svg>` tag and end it there.

**Review by andy-blum — CHANGES_REQUESTED** [review]
Lets remove the `part` attributes from the gradient svg bits, but then I think we're good to go.

**Comment by Valentin-Sorin-Nicolae** [comment]
part="gradient-rect" was removed

**Comment by m4olivei** [comment]
Close / reopen due to [odd failure to build Web Components Storybook](https://dbg-digital-design-team-jenkins.swg-devops.com/job/Carbon%20for%20IBM.com%20-%20Deploy%20Preview%20-%20Web%20Components/11978/console)

File: packages/web-components/src/components/leadspace/leadspace.ts [hunk 2]
```diff
@@ -164,20 +172,20 @@ class C4DLeadSpace extends StableSelectorMixin(LitElement) {
...
-                  class="${c4dPrefix}--leadspace__gradient"
+                  class="${c4dPrefix}--leadspace__gradient" part="image-gradient"
                   viewBox="0 0 100 100"
                   preserveAspectRatio="none"
                   xmlns="http://www.w3.org/2000/svg"
                   xmlns:xlink="http://www.w3.org/1999/xlink"
                 >
                   <defs>
-                    <linearGradient id="stops" class="${c4dPrefix}--leadspace__gradient__stops" gradientTransform="${
+                    <linearGradient id="stops" class="${c4dPrefix}--leadspace__gradient__stops" part="gradient-stop" gradientTransform="${
```
Comment (m4olivei) [inline packages/web-components/src/components/leadspace/leadspace.ts:188]:
```suggestion
                    <linearGradient id="stops" class="${c4dPrefix}--leadspace__gradient__stops" gradientTransform="${
```

File: packages/web-components/src/components/leadspace/leadspace.ts [hunk 3]
```diff
@@ -30,7 +30,14 @@ const { prefix, stablePrefix: c4dPrefix } = settings;
  * @element c4d-leadspace
  * @slot action The action (CTA) content.
  * @slot image The image content.
+ * @csspart content - The content. Usage: `c4d-leadspace::part(content)`
+ * @csspart content-item - The subheading paragraph. Usage: `c4d-leadspace::part(content-item)`
  * @csspart section - The first DOM node inside the shadow-root. Usage: `c4d-leadspace::part(section)`
+ * @csspart container - The container. Usage: `c4d-leadspace::part(container)`
+ * @csspart gradient - The gradient container. Usage: `c4d-leadspace::part(gradient)`
+ * @csspart image-gradient - The image with gradient. Usage: `c4d-leadspace::part(image-gradient)`
+ * @csspart gradient-stop - Where the gradient stops. Usage: `c4d-leadspace::part(gradient-stop)`
```
Comment (m4olivei) [inline packages/web-components/src/components/leadspace/leadspace.ts:39]:
```suggestion
```

**Review by m4olivei — CHANGES_REQUESTED** [review]

**Review by m4olivei — APPROVED** [review]

File: packages/web-components/src/components/leadspace/leadspace.ts [hunk 4]
```diff
@@ -164,13 +171,13 @@ class C4DLeadSpace extends StableSelectorMixin(LitElement) {
     const { gradientStyleScheme, type, size } = this;
     return html`
       <section class="${this._getTypeClass()}" part="section">
-        <div class="${c4dPrefix}--leadspace__container">
-          <div class="${this._getGradientClass()}">
+        <div class="${c4dPrefix}--leadspace__container" part="container">
+          <div class="${this._getGradientClass()}" part="gradient">
```
Comment (m4olivei) [inline packages/web-components/src/components/leadspace/leadspace.ts:175]: This is more accurately `overlay`, then we can use just `gradient` for the actual gradient.
```suggestion
          <div class="${this._getGradientClass()}" part="overlay">
```

File: packages/web-components/src/components/leadspace/leadspace.ts [hunk 5]
```diff
@@ -164,13 +171,13 @@ class C4DLeadSpace extends StableSelectorMixin(LitElement) {
...
             ${gradientStyleScheme === LEADSPACE_GRADIENT_STYLE_SCHEME.NONE
               ? undefined
               : svg`
                 <svg
-                  class="${c4dPrefix}--leadspace__gradient"
+                  class="${c4dPrefix}--leadspace__gradient" part="image-gradient"
```
Comment (m4olivei) [inline packages/web-components/src/components/leadspace/leadspace.ts:180]:
```suggestion
                  class="${c4dPrefix}--leadspace__gradient" part="gradient"
```

**Review by m4olivei — CHANGES_REQUESTED** [review]

**Review by m4olivei — APPROVED** [review]

**Pull request merged**